lint/lenNil_checker.go

```diff
@@ -60,12 +74,17 @@ func (c *lenNilChecker) isRefType(expr ast.Expr) bool {
 	}
 }
 
+func (c *lenNilChecker) isNilKeyword(x ast.Expr) bool {
+	id, ok := x.(*ast.Ident)
+	return ok && id.Name == "nil"
```

Comment:
Could simplify to `qualifiedName(x) == "nil"`.
Also, `nil` is not a keyword. :)